<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ScanForm</class>
 <widget class="QWidget" name="ScanForm">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>593</width>
    <height>415</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_5">
   <item row="0" column="0">
    <widget class="QSplitter" name="splitter">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <widget class="QWidget" name="layoutWidget">
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <item>
        <widget class="QGroupBox" name="groupBox">
         <property name="title">
          <string>Parameters &amp;&amp; Range</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_2">
          <item row="0" column="0">
           <layout class="QVBoxLayout" name="verticalLayout">
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout">
              <item>
               <widget class="QLabel" name="label">
                <property name="toolTip">
                 <string>The parameter to measure.</string>
                </property>
                <property name="text">
                 <string>Observable</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QComboBox" name="observableComboBox">
                <property name="toolTip">
                 <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;The parameter to be measured&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                </property>
                <property name="toolTipDuration">
                 <number>-1</number>
                </property>
                <property name="statusTip">
                 <string/>
                </property>
                <property name="whatsThis">
                 <string/>
                </property>
                <property name="accessibleName">
                 <string/>
                </property>
                <property name="accessibleDescription">
                 <string extracomment="description"/>
                </property>
                <item>
                 <property name="text">
                  <string>UBX Rate</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>GPIO Rate</string>
                 </property>
                </item>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_2">
              <item>
               <widget class="QLabel" name="label_2">
                <property name="toolTip">
                 <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;The Parameter to incrementally change.&lt;/p&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;p&gt;Choose TIME together with &lt;span style=&quot; font-style:italic;&quot;&gt;Nr of Points &lt;/span&gt;and &lt;span style=&quot; font-style:italic;&quot;&gt;Integration Time&lt;/span&gt; to track plot the observable over time.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                </property>
                <property name="text">
                 <string>Scan Parameter</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QComboBox" name="scanParComboBox">
                <property name="toolTip">
                 <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;The independent scan parameter&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                </property>
                <item>
                 <property name="text">
                  <string>THR1</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>THR2</string>
                 </property>
                </item>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_3">
              <item>
               <widget class="QLabel" name="label_3">
                <property name="toolTip">
                 <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Lower and upper boundaries of the scanning parameter range.&lt;/p&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;p&gt;For Threshold scanning, this value is given in Volts [V] (0.020V = 20mV)&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                </property>
                <property name="text">
                 <string>Scan Range from:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="minRangeLineEdit">
                <property name="toolTip">
                 <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Lower scan limit&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                </property>
                <property name="text">
                 <string>0</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="maxRangeOrNrPointsLabel">
                <property name="text">
                 <string>to:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="maxRangeLineEdit">
                <property name="toolTip">
                 <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Upper scan limit&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                </property>
                <property name="text">
                 <string>1</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_8">
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_4">
                <item>
                 <widget class="QLabel" name="label_5">
                  <property name="toolTip">
                   <string>Increment of Scan Parameter points.</string>
                  </property>
                  <property name="text">
                   <string>Step Size</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLineEdit" name="stepSizeLineEdit">
                  <property name="minimumSize">
                   <size>
                    <width>60</width>
                    <height>0</height>
                   </size>
                  </property>
                  <property name="toolTip">
                   <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Scan increment&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                  </property>
                  <property name="text">
                   <string>0.005</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_7">
                <item>
                 <widget class="QCheckBox" name="activateNrMeasurementsCheckBox">
                  <property name="toolTip">
                   <string>Specify number of equally spaced points instead of stepping increments.</string>
                  </property>
                  <property name="text">
                   <string>Nr of Points</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QSpinBox" name="nrMeasurementSpinBox">
                  <property name="enabled">
                   <bool>false</bool>
                  </property>
                  <property name="toolTip">
                   <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Number of scan points&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                  </property>
                  <property name="minimum">
                   <number>1</number>
                  </property>
                  <property name="maximum">
                   <number>10000000</number>
                  </property>
                  <property name="value">
                   <number>10</number>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_5">
              <item>
               <widget class="QLabel" name="label_6">
                <property name="minimumSize">
                 <size>
                  <width>0</width>
                  <height>0</height>
                 </size>
                </property>
                <property name="toolTip">
                 <string>Measurement time interval</string>
                </property>
                <property name="text">
                 <string>Integration Time</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QSpinBox" name="timeIntervalSpinBox">
                <property name="toolTip">
                 <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Measurement duration at each step&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                </property>
                <property name="minimum">
                 <number>1</number>
                </property>
                <property name="maximum">
                 <number>10000</number>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="label_7">
                <property name="text">
                 <string>s</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_6">
              <item>
               <widget class="QCheckBox" name="limitStatisticsCheckBox">
                <property name="toolTip">
                 <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Skip to the next scan point when a predefined statistics is satisfied&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                </property>
                <property name="text">
                 <string>limit statistics</string>
                </property>
               </widget>
              </item>
              <item>
               <spacer name="horizontalSpacer">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item>
               <widget class="QLabel" name="label_8">
                <property name="text">
                 <string>max counts:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QComboBox" name="maxStatisticsComboBox">
                <property name="enabled">
                 <bool>false</bool>
                </property>
                <property name="toolTip">
                 <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Skip to next scan step when this number of events is recorded&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                </property>
                <property name="currentIndex">
                 <number>2</number>
                </property>
                <item>
                 <property name="text">
                  <string>10</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>100</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>1k</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>10k</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>100k</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>1M</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>10M</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>100M</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>1G</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>10G</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>100G</string>
                 </property>
                </item>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <layout class="QGridLayout" name="gridLayout_3">
         <item row="0" column="0">
          <spacer name="verticalSpacer">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>18</height>
            </size>
           </property>
          </spacer>
         </item>
         <item row="1" column="0">
          <widget class="QPushButton" name="scanStartPushButton">
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Start or stop the scan process&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="text">
            <string>Start Scan</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QGroupBox" name="currentScanGroupBox">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="title">
          <string>Current Scan</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_4">
          <item row="0" column="0">
           <layout class="QVBoxLayout" name="verticalLayout_2">
            <item>
             <widget class="QLabel" name="scanparNameLabel">
              <property name="text">
               <string>Scan Par</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignCenter</set>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="currentScanparLabel">
              <property name="font">
               <font>
                <pointsize>11</pointsize>
               </font>
              </property>
              <property name="frameShape">
               <enum>QFrame::Panel</enum>
              </property>
              <property name="text">
               <string>N/A</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignCenter</set>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="0" column="1">
           <spacer name="horizontalSpacer_2">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>93</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="0" column="2">
           <layout class="QVBoxLayout" name="verticalLayout_3">
            <item>
             <widget class="QLabel" name="observableNameLabel">
              <property name="text">
               <string>Observable</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignCenter</set>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="currentObservableLabel">
              <property name="font">
               <font>
                <pointsize>11</pointsize>
               </font>
              </property>
              <property name="frameShape">
               <enum>QFrame::Panel</enum>
              </property>
              <property name="text">
               <string>N/A</string>
              </property>
              <property name="scaledContents">
               <bool>false</bool>
              </property>
              <property name="alignment">
               <set>Qt::AlignCenter</set>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="0" column="3">
           <spacer name="horizontalSpacer_3">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>93</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QProgressBar" name="scanProgressBar">
         <property name="value">
          <number>0</number>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QFrame" name="frame">
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <layout class="QGridLayout" name="gridLayout">
       <item row="0" column="0" colspan="2">
        <widget class="CustomPlot" name="scanPlot" native="true"/>
       </item>
       <item row="1" column="0">
        <widget class="QCheckBox" name="plotDifferentialCheckBox">
         <property name="toolTip">
          <string>Plot the difference between subsequent data points</string>
         </property>
         <property name="text">
          <string>differential</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QPushButton" name="exportDataPushButton">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Export measurements to file&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="text">
          <string>export data</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>CustomPlot</class>
   <extends>QWidget</extends>
   <header>custom_plot_widget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>limitStatisticsCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>maxStatisticsComboBox</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>83</x>
     <y>176</y>
    </hint>
    <hint type="destinationlabel">
     <x>295</x>
     <y>176</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>activateNrMeasurementsCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>nrMeasurementSpinBox</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>218</x>
     <y>153</y>
    </hint>
    <hint type="destinationlabel">
     <x>323</x>
     <y>153</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>activateNrMeasurementsCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>stepSizeLineEdit</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>218</x>
     <y>153</y>
    </hint>
    <hint type="destinationlabel">
     <x>125</x>
     <y>153</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
